src/sql/parts/query/editor/queryResultsView.ts

```diff
@@ -223,6 +226,23 @@ export class QueryResultsView extends Disposable {
 				this._panelView.pushTab(this.topOperationsTab);
 			}
 		}
+
+		// restore query model view tabs
+		this.input.state.visibleTabs.forEach(tabId => {
+			if (tabId.startsWith('querymodelview;')) {
+				// tab id format is 'tab type;title;model view id'
+				let parts = tabId.split(';');
+				if (parts.length === 3) {
+					let tab = this._register(new QueryModelViewTab(parts[1], this.instantiationService));
```

Comment:
Are tabs tied to inputs? If so they should be registered in the inputs so they are disposed with the inputs not the results view

Reply:
Tabs are tied to a given resultset, the input's lifetime is longer than tabs.